cvmfs/server/cvmfs_server_ingest_tarball.sh

```diff
@@ -0,0 +1,326 @@
+#
+# This file is part of the CernVM File System
+# This script takes care of creating, removing, and maintaining repositories
+# on a Stratum 0/1 server
+#
+# Implementation of the "cvmfs_server ingest-tarball" command
+
+# This file depends on fuctions implemented in the following files:
+# - cvmfs_server_util.sh
+# - cvmfs_server_common.sh
+
+
+# TODO Most of this code is replicated and shared between different scrips,
+# it would be a good idea to refactor common patterns into coherent functions.
+
+cvmfs_server_ingest_tarball() {
```

Comment:
Do you mean like what we do with the normal publish. `cvmfs_server transaction ...` then the `cvmfs_server ingest-tarball ...` and finally `cvmfs_server publish ...` or simply open a transaction inside the `cvmfs_server ingest-tarball` command?

Reply:
I was thinking of the latter: a call to `cvmfs_server ingest` would call `cvmfs_server_transaction` at the beginning and `cvmfs_server_publish` or `cvmfs_server_abort` at the end.  Between these calls, it uses `cvmfs_swissknife ingest` in order to process a tarball instead of the normal `sync`.